src/magic_folder/web.py

```diff
@@ -264,41 +280,38 @@ def add_participant(self, request, folder_name):
             returnValue(NoResource(b"{}"))
 
         body = request.content.read()
-        try:
-            participant = json.loads(body)
-            required_keys = {
-                "author",
-                "personal_dmd",
-            }
-            required_author_keys = {
-                "name",
-                # not yet
-                # "public_key_base32",
-            }
-            if set(participant.keys()) != required_keys:
-                raise _InputError("Require input: {}".format(", ".join(sorted(required_keys))))
-            if set(participant["author"].keys()) != required_author_keys:
-                raise _InputError("'author' requires: {}".format(", ".join(sorted(required_author_keys))))
-
-            author = create_author(
-                participant["author"]["name"],
-                # we don't yet properly track keys but need one
-                # here .. this won't be correct, but we won't use
-                # it .. following code still only looks at the
-                # .name attribute
-                # see https://github.com/LeastAuthority/magic-folder/issues/331
-                VerifyKey(os.urandom(32)),
-            )
+        participant = _load_json(body)
+
+        required_keys = {
+            "author",
+            "personal_dmd",
+        }
+        required_author_keys = {
+            "name",
+            # not yet
+            # "public_key_base32",
+        }
+        if set(participant.keys()) != required_keys:
+            raise _InputError("Require input: {}".format(", ".join(sorted(required_keys))))
+        if set(participant["author"].keys()) != required_author_keys:
+            raise _InputError("'author' requires: {}".format(", ".join(sorted(required_author_keys))))
+
+        author = create_author(
+            participant["author"]["name"],
+            # we don't yet properly track keys but need one
+            # here .. this won't be correct, but we won't use
+            # it .. following code still only looks at the
+            # .name attribute
+            # see https://github.com/LeastAuthority/magic-folder/issues/331
+            VerifyKey(os.urandom(32)),
+        )
 
-            dmd = tahoe_uri_from_string(participant["personal_dmd"])
-            if not IDirnodeURI.providedBy(dmd):
-                raise _InputError("personal_dmd must be a directory-capability")
-            if not dmd.is_readonly():
-                raise _InputError("personal_dmd must be read-only")
-            personal_dmd_cap = participant["personal_dmd"]
-        except _InputError as e:
-            request.setResponseCode(http.BAD_REQUEST)
-            returnValue(json.dumps({"reason": str(e)}))
+        dmd = tahoe_uri_from_string(participant["personal_dmd"])
+        if not IDirnodeURI.providedBy(dmd):
```

Comment:
I know this is just re-indented existing code, but there's `util.capabilitiies.*` methods (for future reference).